azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -41,15 +65,118 @@ def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_ev
         else:
             logger.error(msg)
         if send_event:
-            add_event(op=op, message=msg, is_success=is_success)
+            add_event(op=op, message=msg, is_success=is_success, log_event=False)
 
     @staticmethod
     def is_policy_enforcement_enabled():
         """
-        Check whether user has opted into policy enforcement feature.
+        Policy will be enabled if (1) policy file exists at CUSTOM_POLICY_PATH and (2) the conf flag "Debug.EnableExtensionPolicy" is true.
         Caller function should check this before performing any operations.
         """
-        # TODO: Add check for policy file present at /etc/waagent_policy.json.
         # Policy should only be enabled if conf flag is true AND policy file is present.
-        return conf.get_extension_policy_enabled()
+        policy_file_exists = os.path.exists(CUSTOM_POLICY_PATH)
+        return conf.get_extension_policy_enabled() and policy_file_exists
+
+    def __get_policy(self):
+        """
+         Check if custom policy exists at CUSTOM_POLICY_PATH, load JSON object and return as a dict.
+        Return {} if no policy exists.
+        The expected policy format is:
+         {
+            "policyVersion": str,
+            "extensionPolicies": {
+                "allowListedExtensionsOnly": bool,
+                "signatureRequired": bool,
+                "extensions": {
+                    "<extensionName1>": {
+                        "signatureRequired": bool
+                    }
+                }
+            }
+        }
+        """
+        # TODO: Add schema validation for custom policy file and raise relevant error message to user.
+        if os.path.exists(CUSTOM_POLICY_PATH):
+            self._log_policy("Custom policy found at {0}. Using custom policy.".format(CUSTOM_POLICY_PATH))
+            with open(CUSTOM_POLICY_PATH, 'r') as f:        # Open file in read-only mode.
+                custom_policy = json.load(f)
```

Comment:
What happens if policy file content is not valid json? Does this raise exception? How do we want to handle that right now

Reply:
Replied to your other comment on this issue, resolving this one.